packages/node_modules/samples/browser-multi-party-call/test/wdio/spec/dial-before-connect.js

```diff
@@ -0,0 +1,114 @@
+
+import '@ciscospark/plugin-phone';
+
+import CiscoSpark from '@ciscospark/spark-core';
+import testUsers from '@ciscospark/test-helper-test-users';
+import setValue from '../lib/set-value';
+
+describe(`sample-browser-multi-party-call`, () => {
+  describe(`dial before connecting`, () => {
+    let mccoy, roomId, spock;
+
+    const browserSpock = browser.select(`browserSpock`);
+    const browserMccoy = browser.select(`browserMccoy`);
+
+    const users = {
+      mccoy: null,
+      spock: null
+    };
+
+    before(`create users and register`, () => testUsers.create({count: Object.keys(users).length})
+      .then((created) => Promise.all(Object.keys(users).map((name, index) => {
+        const user = users[name] = created[index];
+        user.spark = new CiscoSpark({
+          credentials: {
+            authorization: user.token
+          }
+        });
+
+        console.log(user.spark);
+
+        return user.spark.phone.register();
+      })))
+      .then(() => {
+        mccoy = users.mccoy;
+        spock = users.spock;
+      }));
+
+    beforeEach(`enable group calling`, () => {
+      spock.spark.config.phone.enableExperimentalGroupCallingSupport = true;
+      mccoy.spark.config.phone.enableExperimentalGroupCallingSupport = true;
+    });
+
+    afterEach(`disable group calling`, () => {
+      spock.spark.config.phone.enableExperimentalGroupCallingSupport = false;
+      mccoy.spark.config.phone.enableExperimentalGroupCallingSupport = false;
+    });
+
+    step(`create test room and add members to room`, () => {
+      spock.spark.request({
+        method: `POST`,
+        service: `hydra`,
+        resource: `rooms`,
+        body: {
+          title: `Call Test`
+        }
+      })
+        .then((res) => {
+          const room = res.body;
+
+          roomId = room.id;
+
+          return spock.spark.request({
+            method: `POST`,
+            service: `hydra`,
+            resource: `memberships`,
+            body: {
+              roomId: room.id,
+              personId: mccoy.id
+            }
+          });
+        });
+    });
+
+    step(`loads the app`, () => {
+      browser.reload();
+      browser.url(`/browser-multi-party-call/`);
+    });
+
+    step(`places call from spock to mccoy`, () => {
+      browserSpock.waitUntil(() => browserSpock.getTitle() === `Sample: Multi Party Calling`);
+      browserSpock.execute(setValue, `[placeholder="Your access token"]`, spock.token.access_token);
+      browserSpock.setValue(`[name="room"]`, roomId);
+      browserSpock.click(`[title="dial"]`);
+
+      browserSpock.waitForExist(`.listening`);
+    });
+
+    step(`connects mccoy's browser`, () => {
+      browserMccoy.waitUntil(() => browserMccoy.getTitle() === `Sample: Multi Party Calling`);
+      browserMccoy.execute(setValue, `[placeholder="Your access token"]`, mccoy.token.access_token);
+      browserMccoy.click(`[title="connect"]`);
+
+      // TODO how do we wait for a modal to open?
```

Comment:
You could wait for [#alertText](http://webdriver.io/api/protocol/alertText.html) to return a value, but doesn't seem to do much more than what you are doing now.